tests/test_span_session.rb

```diff
@@ -0,0 +1,99 @@
+require_relative 'ciscotest'
+require_relative '../lib/cisco_node_utils/span_session'
+require_relative '../lib/cisco_node_utils/interface'
+require_relative '../lib/cisco_node_utils/vlan'
+require_relative '../lib/cisco_node_utils/cisco_cmn_utils'
+
+include Cisco
+
+# TestSpanSession - Minitest for SPAN session node utility
+class TestSpanSession < CiscoTestCase
+  @skip_unless_supported = 'session'
+  @@cleaned = false # rubocop:disable Style/ClassVars
+
+  def setup
+    super
+    cleanup unless @@cleaned
+    @@cleaned = true # rubocop:disable Style/ClassVars
+  end
+
+  def teardown
+    cleanup
+  end
+
+  def cleanup
+    SpanSession.sessions.each do |_session, obj|
+      obj.destroy
+    end
+  end
+
+  def test_create_new_session
+    span = SpanSession.new(1)
+
+    assert_equal(span.session_id, 1)
+    assert_equal(span.type, 'local') # default session type
+    assert(span.shutdown)
+    span.destroy
+  end
+
+  def test_remove_session
+    span = SpanSession.new(1)
+    span.destroy
+    refute(span, 'Session was not cleaned up correctly...')
+  end
+
+  def test_create_session_invalid_id
+    e = assert_raises(CliError) { SpanSession.new(33) }
+    assert_match(/Invalid value.range/, e.message)
+  end
+
+  def test_session_type
+    span = SpanSession.new(1)
+    erspan_type = 'erspan-source'
+    span.type = erspan_type
```

Comment:
I tried running this test but I get the following error:
```
TestSpanSession#test_session_type:
Cisco::CliError: [#<Cisco::SpanSession:0x007f49f6d16450>] The command 'monitor session 1 type erspan-source' was rejected with error:
ERROR: Invalid ssn type specified
```
You will likely need to remove the session id it it exists and then reconfigure it when changing the type.